src/oauth2c_token_cache.erl

```diff
@@ -0,0 +1,155 @@
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%%% @doc OAuth2 Authentication Token Cache - This gen_server implements a
+%%% simple caching mechanism for authentication tokens.
+%%% @end
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+%%%_* Module declaration ===============================================
+-module(oauth2c_token_cache).
+-behaviour(gen_server).
+
+%%%_* Exports ==========================================================
+
+-export([start/0]).
+-export([start/1]).
+-export([start_link/0]).
+-export([start_link/1]).
+-export([get/1]).
+-export([set_and_get/2]).
+-export([set_and_get/3]).
+
+%% gen_server
+-export([init/1]).
+-export([handle_call/3]).
+-export([handle_cast/2]).
+-export([clear/0]).
+
+%%%_* Macros =========================================================
+
+-define(DEFAULT_TTL, 3600). % Default cache entry TTL in seconds.
+-define(SERVER, ?MODULE).
+-define(TOKEN_CACHE_ID, token_cache_id).
+
+%%%_* Code =============================================================
+%%%_ * Types -----------------------------------------------------------
+%%%_ * API -------------------------------------------------------------
+
+-spec start() -> {atom(), pid()}.
+start() ->
+  start(?DEFAULT_TTL).
+-spec start(non_neg_integer()) -> {atom(), pid()}.
+start(DefaultTTL) ->
+  gen_server:start({local, ?SERVER}, ?SERVER,
+                   #{default_ttl => DefaultTTL}, []).
+
+-spec start_link() -> {atom(), pid()}.
+start_link() ->
+  start_link(?DEFAULT_TTL).
+-spec start_link(non_neg_integer()) -> {atom(), pid()}.
+start_link(DefaultTTL) ->
+  gen_server:start_link({local, ?SERVER}, ?SERVER,
+                        #{default_ttl => DefaultTTL}, []).
+
+-spec get(integer()) -> {error, atom()} | {ok, term()}.
+get(Key) ->
+  get_token(Key).
+
+-spec set_and_get(Key, LazyValue) -> Value | Error when
+    Key :: integer(),
+    LazyValue :: fun(() -> Value | Error),
+    Value :: {ok, term()},
+    Error :: {error, binary()}.
+set_and_get(Key, LazyValue) ->
+  set_and_get(Key, LazyValue, []).
+
+-spec set_and_get(Key, LazyValue, Options) -> Value | Error when
+    Key :: integer(),
+    LazyValue :: fun(() -> Value | Error),
+    Options :: list(),
+    Value :: {ok,  term()},
+    Error :: {error, atom()}.
+set_and_get(Key, LazyValue, Options) ->
```

Comment:
I think adding options here for one option is unnecessary. `CurrentExpiryTime` would be a good name.

Reply:
You're right, I have replaced options with just an integer called  `CurrentExpiryTime `.